tests/integration/test_api/test_config/test_logs/test_api_logs_formats.py

```diff
@@ -0,0 +1,190 @@
+    """Match a given line with a regular expression."""
+
+    msg = fr'(\d+/\d+/\d+ \d+:\d+:\d+) INFO: wazuh 127.0.0.1 "GET {API_LOGIN_ENDPOINT}" with parameters' \
+          r' {} and body {} done in (\d*\.*\d+)s: (\d+)'
+    match = re.match(msg, line)
+    return match
+
+
+def callback_json_log_error(line):
+    """Match a given line with a regular expression."""
+
+    msg = r'{"timestamp": "(\d+/\d+/\d+ \d+:\d+:\d+)", "levelname": "ERROR", "data": {"type": "informative", ' \
+          fr'"payload": "{error_log_payload}"}}'
+    match = re.match(msg, line)
+    return match
+
+
+def callback_plain_error(line):
+    """Match a given line with a regular expression."""
+
+    msg = fr"(\d+/\d+/\d+ \d+:\d+:\d+) ERROR: {error_log_payload}"
+    match = re.match(msg, line)
+    return match
+
+
+# Tests
+@pytest.mark.filterwarnings('ignore::urllib3.exceptions.InsecureRequestWarning')
+def test_api_logs_formats(get_configuration, configure_api_environment, clean_log_files, daemons_handler,
+                          wait_for_start):
+    '''
+    description: The test checks if the logs of the API are stored in the specified formats and the content of the log
+                 files are the expected.
+    wazuh_min_version: 4.4.0
+    parameters:
+        - get_configuration:
+            type: fixture
+            brief: Get configurations from the module.
+        - configure_api_environment:
+            type: fixture
+            brief: Configure a custom environment for API testing.
+        - clean_log_files:
+            type: fixture
+            brief: Reset the log files of the API and delete the rotated log files.
+        - daemons_handler:
+            type: fixture
+            brief: Handle the Wazuh daemons.
+        - wait_for_start:
+            type: fixture
+            brief: Wait until the API starts.
+    assertions:
+        - Verify that the expected log exists in the log file.
+        - Verify that the logs values are the same in both log formats.
+    input_description: The test gets the configuration from the YAML file, which contains the API configuration.
+    expected_output:
+        - The log was not the expected.
+        - The length of the subgroups of the match is not equal.
+        - The values of the logs doesn't match.
+    tags:
+        - api
+        - logs
+        - logging
+    '''
+    wazuh_log_monitor = FileMonitor(API_LOG_FILE_PATH)
+    json_log_monitor = FileMonitor(API_JSON_LOG_FILE_PATH)
+
+    current_formats = get_configuration['configuration']['logs']['format'].split(',')
+
+    send_request()
+
+    expected_error =  'expected_error' in get_configuration
+    if 'json' in current_formats:
+        callback = callback_json_log_error if expected_error else callback_json_log_login_info
+        json_result = json_log_monitor.start(timeout=API_GLOBAL_TIMEOUT, callback=callback,
+                                             error_message='The log was not the expected.').result()
```

Comment:
Improve the file monitor error message. Something like this will do the job:
```suggestion
                                             error_message='JSON API error log was not been generated .').result()
```